include/ignition/rendering/base/BaseScene.hh

```diff
@@ -55,9 +55,11 @@ namespace ignition
 
       public: virtual std::string Name() const override;
 
-      public: virtual common::Time SimTime() const override;
+      public: virtual std::chrono::steady_clock::time_point SimTime()
+        const override;
 
-      public: virtual void SetSimTime(const common::Time &_time) override;
+      public: virtual void SetSimTime(
+        const std::chrono::steady_clock::time_point &_time) override;
```

Comment:
All functions in the public API should go through a deprecation cycle (deprecate on `ign-rendering4` and remove on `ign-rendering5`).

We can't overload the `SimTime()` function, so this is tricky. I don't know what @iche033 would prefer - no tick-tock, or a change in the function name.

Reply:
friendly ping @iche033

Reply:
How about changing the functions to `SetTime` and `Time`? Technically `ign-rendering` doesn't care about whether it's displaying a simulation or something else.

Reply:
I think the `SetSimTime` and `SimTime` APIs were added a while back mainly for the [gazebo_scene_viewer](https://github.com/ignitionrobotics/ign-rendering/blob/master/examples/gazebo_scene_viewer/SceneManager.cc#L862) example. It's not really used internally. Changing them to `SetTime` and `Time` sounds good.

Reply:
as for the tricky situation with changing return type, I remember in the past we would overload the `operator=` function in return type class, i.e. in this case, we overload the `Time::operator=` to take in a `std::chrono` object. Now that we're just changing the functions to `SetTime` and `Time`, we don't have to do this but I thought to bring it up if that's needed somewhere else.